pkg/ddl/multi_schema_change.go

```diff
@@ -373,30 +395,47 @@ func checkOperateSameColAndIdx(info *model.MultiSchemaInfo) error {
 	return checkIndexes(info.AlterIndexes, true)
 }
 
-func mergeAddIndex(info *model.MultiSchemaInfo) {
-	var mergedSubJob *model.SubJob
-	var mergeCnt int
+func mergeAddIndex(
```

Comment:
maybe we can calculate a appropriate `LocalWriterMemCacheSize` for the job when run it.

Reply:
I can add a TODO for it. To adjust the writer count, LocalWriterMemCacheSize and other things based on resource limit.

Reply:
i mean to use `adjust LocalWriterMemCacheSize and other things based on resource limit.` to replace current solution, and if there are NO minimal resource required, report error

> To adjust the writer count

better adjust by user

Reply:
The background of this PR is, in a multi-schema change of many ADD INDEXes, TiDB gets OOM and seems lightning's `TotalMemoryConsume` is underestimated. So I gradually test memory usage in https://github.com/pingcap/tidb/pull/53856. And after `TotalMemoryConsume` return a larger value, a CI failed because the multi-schema change has exceeded the memory limit. So my first idea is to limit the merged group size of multi-schema change.

> i mean to use adjust LocalWriterMemCacheSize and other things based on resource limit. to replace current solution,

I'll consider it now. Maybe I should write more details in an issue and we can decide to choose which solution.